pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -75,6 +88,9 @@ func (pl *NodeAffinity) Score(ctx context.Context, state *framework.CycleState,
 	affinity := pod.Spec.Affinity
 
 	var count int64
+	if pl.defPrefSchedTerms != nil {
+		count += pl.defPrefSchedTerms.Score(node)
+	}
```

Comment:
If the pod specified a same prefered affinity with default one, then that will be double count. For example, the default one is 'in zoneA with weight=1', the pod specified one is 'in zoneA with weight=1 and in zoneB with wieght=2'. In the end, actually zoneA and zoneB have the same weight. That changed the original intent for the user. But this is indeed the `AND` behavior as the design, just thinking about whether this makes sense for the user. Although I see the major use-case for this feature is required affinity, so I may over thing about this.

Reply:
You are correct, and I think that is working as intended.

However, in "AddedAffinity", you would probably use a label that Pods usually don't care about. I'll try to put some thoughts in kubernetes/website#24914